django_elasticsearch_dsl/signals.py

```diff
@@ -96,6 +99,124 @@ def teardown(self):
         models.signals.m2m_changed.disconnect(self.handle_m2m_changed)
         models.signals.pre_delete.disconnect(self.handle_pre_delete)
 
+try:
+    from celery import shared_task
+except ImportError:
+    pass
+else:
+    class CelerySignalProcessor(RealTimeSignalProcessor):
+        """Celery signal processor.
+
+        Allows automatic updates on the index as delayed background tasks using
+        Celery.
+
+        NB: We cannot process deletes as background tasks.
+        By the time the Celery worker would pick up the delete job, the
+        model instance would already deleted. We can get around this by
+        setting Celery to use `pickle` and sending the object to the worker,
+        but using `pickle` opens the application up to security concerns.
+        """
 
-# Sent after document indexing is completed
-post_index = Signal()
+        def handle_save(self, sender, instance, **kwargs):
+            """Handle save with a Celery task.
+
+            Given an individual model instance, update the object in the index.
+            Update the related objects either.
+            """
+            pk = instance.pk
+            app_label = instance._meta.app_label
+            model_name = instance.__class__.__name__
+
+            self.registry_update_task.delay(pk, app_label, model_name)
```

Comment:
It is better to use [`transaction.on_commit`](https://docs.djangoproject.com/en/4.2/topics/db/transactions/#django.db.transaction.on_commit) hook for this kind of thing. Otherwise, if the model is saved inside a atomic transaction, The task will not be able to fetch the object.

Reply:
It has been changed, but it seems that on_commit is not triggered in django's TestCase, and I am still thinking of a suitable way to reuse test cases